src/NuGet.Core/NuGet.Frameworks/NuGetFrameworkFactory.cs

```diff
@@ -70,6 +70,129 @@ public static NuGetFramework Parse(string folderName, IFrameworkNameProvider map
             return framework;
         }
 
+        /// <summary>
+        /// Creates a NuGetFramework from individual components
+        /// </summary>
+        public static NuGetFramework ParseComponents(string targetFrameworkIdentifier, string targetFrameworkVersion, string targetFrameworkProfile, string targetPlatformIdentifier, string targetPlatformVersion)
+        {
+            return ParseComponents(targetFrameworkIdentifier, targetFrameworkVersion, targetFrameworkProfile, targetPlatformIdentifier, targetPlatformVersion, DefaultFrameworkNameProvider.Instance);
+        }
+
+        /// <summary>
+        /// Creates a NuGetFramework from individual components, using the given mappings
+        /// </summary>
+        public static NuGetFramework ParseComponents(string targetFrameworkIdentifier, string targetFrameworkVersion, string targetFrameworkProfile, string targetPlatformIdentifier, string targetPlatformVersion, IFrameworkNameProvider mappings)
+        {
+            if (targetFrameworkIdentifier == null)
```

Comment:
What about empty length string?
```
if (string.IsNullOrEmpty(targetFrameworkIdentifier))
{
    throw new ArgumentException(Strings.ArgumentNullOrEmpty, nameof(targetFrameworkIdentifier));
}
```

Like [this](https://github.com/NuGet/NuGet.Client/blob/886475792e69b0567173a658c7d7cc79bfa87228/src/NuGet.Core/NuGet.Packaging/Signing/TrustedSigners/TrustedSignersProvider.cs#L37).